packages/devtools_app/lib/src/screens/debugger/debugger_controller.dart

```diff
@@ -480,38 +486,38 @@ class DebuggerController extends DisposableController
   /// Call `toString()` on the given instance and return the result.
   Future<Response> invokeToString(InstanceRef instance) {
     return _service.invoke(
-      isolateRef.id,
-      instance.id,
+      isolateRef.id!,
+      instance.id!,
       'toString',
       <String>[],
       disableBreakpoints: true,
     );
   }
 
   /// Retrieves the full string value of a [stringRef].
-  Future<String> retrieveFullStringValue(
+  Future<String?> retrieveFullStringValue(
     InstanceRef stringRef, {
-    String onUnavailable(String truncatedValue),
+    String onUnavailable(String? truncatedValue)?,
   }) async {
-    return serviceManager.service.retrieveFullStringValue(
-      isolateRef.id,
+    return _service.retrieveFullStringValue(
+      isolateRef.id!,
       stringRef,
       onUnavailable: onUnavailable,
     );
   }
 
   Future<void> clearBreakpoints() async {
-    final breakpoints = _breakpoints.value.toList();
-    await Future.forEach(breakpoints, (Breakpoint breakpoint) {
-      return removeBreakpoint(breakpoint);
+    final breakpoints = _breakpoints.value!.toList();
+    await Future.forEach(breakpoints, (Breakpoint? breakpoint) {
```

Comment:
can breakpoint be non-nulalble? when would we have a null value in the breakpoints list